GillianCore/debugging/debugger/debugger.ml

```diff
@@ -167,7 +173,12 @@ struct
           let asrt, _ = asrt_report.step in
           Fmt.str "%a" Asrt.pp asrt
         in
-        let substitutions = asrt_report.subst |> Subst.to_list_pp in
+        let substitutions =
+          asrt_report.subst |> Subst.to_list_pp
+          |> List.map (fun subst ->
+                 List.find_opt (fun prev -> prev.subst = subst) prev_substs
```

Comment:
Let's not use polymorphic equal
`[%eq: (string * string)] prev.subst subst` should work